From 29eb97cf1defd6fce90fe5db5df4879663aceea6 Mon Sep 17 00:00:00 2001
From: Gert <Gert-dev@users.noreply.github.com>
Date: Thu, 27 Jan 2022 20:38:08 +0100
Subject: [PATCH] FIX(client): Fix PipeWire not being usable in Flatpaks

PipeWire is correctly built and enabled in a Flatpak runtime, but does
not properly intialize because the .so filename of the library is named
slightly differently there.

See also https://github.com/flathub/info.mumble.Mumble/issues/19#issuecomment-1023001248
---
 src/mumble/PipeWire.cpp | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/mumble/PipeWire.cpp b/src/mumble/PipeWire.cpp
index 438d24b82e..ed03cd0313 100644
--- a/src/mumble/PipeWire.cpp
+++ b/src/mumble/PipeWire.cpp
@@ -128,7 +128,15 @@ void PipeWireInit::destroy() {
 static PipeWireInit pwi;
 
 PipeWireSystem::PipeWireSystem() : m_ok(false), m_users(0) {
-	const QStringList names{ "libpipewire.so", "libpipewire-0.3.so" };
+	// clang-format off
+	const QStringList names {
+		// Common names used by Linux distributions.
+		"libpipewire.so",
+		"libpipewire-0.3.so",
+		// Name used by the Flatpak FreeDesktop runtime.
+		"libpipewire-0.3.so.0"
+	};
+	// clang-format on
 
 	for (const auto &name : names) {
 		m_lib.setFileName(name);
